### PR TITLE
Run Terraform plan when TF config changes in PR

### PR DESCRIPTION
**Pull request by ThePletch**
* Automatically runs Terraform plans for each environment each time a commit lands in a PR that changes our TF config, and adds a comment to the PR with the results of the plan
* Reorganizes our TF code to use relative paths to templates/modules instead of relying on a root directory mount in Docker
* Sets up centralized credentials in Vault for use in CI/automations that interact with Terraform
* Bumps our version lock for the Terraform Google provider to the latest version to account for some features being moved from google-beta to GA.

### CODE REVIEW

**Comment by ThePletch** [comment]
that's a whole lot of comments

File: .github/workflows/terraformPlan.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          terraform_version: ${{ env.terraform_version }}
+
+      - name: "${{ matrix.environment }} set VAULT_TOKEN"
+        id: token
+        run: |
+          VAULT_TOKEN=$(curl \
+            --request POST \
+            --data '{"role_id":"'"${ROLE_ID}"'","secret_id":"'"${SECRET_ID}"'"}' \
+            ${VAULT_ADDR}/v1/auth/approle/login | jq -r .auth.client_token)
+          echo ::add-mask::${VAULT_TOKEN}
```
Comment (aherbst-broad) [inline .github/workflows/terraformPlan.yml:65]: 👍 til about ::add-mask in github workflows

Comment (ThePletch) [reply .github/workflows/terraformPlan.yml:65]: yeah the ability to dynamically mark things as sensitive is really great

**Review by aherbst-broad — APPROVED** [review]
This seems reasonable to my unskilled terraform eyes.

**Review by raaidbroad — APPROVED** [review]
lgtm

**Pull request merged**